ros2doctor/ros2doctor/api/__init__.py

```diff
@@ -85,10 +85,13 @@ def run_checks() -> Tuple[Set[str], int, int]:
             result = check_instance.check()
         except Exception:
             doctor_warn('Fail to call %s class functions.' % check_entry_pt.name)
-        if result is False:
-            fail += 1
-            failed_cats.add(check_category)
-        total += 1
+        if isinstance(result, bool):
+            if result is False:
+                fail += 1
+                failed_cats.add(check_category)
+            total += 1
+        else:
+            doctor_warn('Check result from %s is invalid.' % check_entry_pt.name)
```

Comment:
If the plugin is mis-implemented, should we consider that a failure and thus increment `fail`, `total`, and add it to `failed_cats`?  Or add a new return tuple, `broken_cats`?

Reply:
`failed_cats` shows report for failed checks only and makes it easier for users to look into ROS related issues. Implementation mistakes are unrelated to ROS and I'd prefer to keep it separated from check failures. `broken_cats` is a feasible solution, but I think current warning and entry point name give enough info for developers to debug. Having a specific structure would seem like we expect lots of implementation errors, and would lead to a feature request like "debug-code-for-me" rabbit hole.

Reply:
I'd like to keep implementation errors separated from check failures for now as they are unrelated to ROS system issues. Entry name and warning msg give enough info to help debug. I don't expect a lot of plugin errors yet to justify a specific return structure. Good suggestion on `broken_cats` though, I'll keep it in mind for future needs.

Reply:
That's OK as it goes.  To be honest, I wouldn't try to detect broken plugins at all; that's up to the plugin authors to get correct.  If they get it wrong, things won't work, but that's the same thing that happens across the entire codebase.

To be more concrete: I would just remove the whole `if isinstance(result, bool):` block, I just don't think it adds much value.